modules/core-components/src/test/java/org/mule/runtime/core/internal/source/polling/DefaultSchedulerMessageSourceTestCase.java

```diff
@@ -126,6 +129,80 @@ public String describeFailure() {
     });
   }
 
+  @Test
+  public void testStartWhenStopping() throws Exception {
+    DefaultSchedulerMessageSource schedulerMessageSource = createMessageSource();
+    SensingNullMessageProcessor flow = getSensingNullMessageProcessor();
+    schedulerMessageSource.setListener(flow);
+    schedulerMessageSource.setAnnotations(singletonMap(LOCATION_KEY, TEST_CONNECTOR_LOCATION));
+
+    DefaultMuleContext spyMuleContext = spy((DefaultMuleContext) muleContext);
+    doReturn(true).when(spyMuleContext).isStopping();
+
+    schedulerMessageSource.setMuleContext(spyMuleContext);
+    schedulerMessageSource.start();
+
+    assertThat("Scheduler should not be started when MuleContext is stopping",
+               schedulerMessageSource.isStarted(), is(false));
+  }
+
+  @Test
+  public void testRunWhenStopping() throws Exception {
+    DefaultSchedulerMessageSource schedulerMessageSource = createMessageSource();
+    SensingNullMessageProcessor flow = getSensingNullMessageProcessor();
+    schedulerMessageSource.setListener(flow);
+    schedulerMessageSource.setAnnotations(singletonMap(LOCATION_KEY, TEST_CONNECTOR_LOCATION));
+
+    DefaultMuleContext spyMuleContext = spy((DefaultMuleContext) muleContext);
+    doReturn(false).when(spyMuleContext).isStopping();
+    doReturn(true).when(spyMuleContext).isPrimaryPollingInstance();
+
+    schedulerMessageSource.setMuleContext(spyMuleContext);
+    schedulerMessageSource.start();
+    doReturn(true).when(spyMuleContext).isStopping();
+    new PollingProber(RECEIVE_TIMEOUT, 100).check(new Probe() {
+
+      @Override
+      public boolean isSatisfied() {
+        return flow.event == null;
```

Comment:
I'm not sure why you're asserting this in a prober. The probers are used to check a condition that "becomes" true asynchronously. In this test, isn't the `flow.event` always null?
In the other hand, I believe that the other two verify() that you do at the end of the test are enough to test that the message is not being processed (please add a comment saying that the source won't call isPrimaryPollingInstance because it happens in the run method, and you're asserting it because of that)